cmd/operator/main.go

```diff
@@ -18,15 +18,56 @@ package main
 
 import (
 	"flag"
+	"fmt"
 	"os"
 
 	"github.com/rhobs/observability-operator/pkg/operator"
 	"go.uber.org/zap/zapcore"
 
+	obopo "github.com/rhobs/obo-prometheus-operator/pkg/operator"
+
+	k8sflag "k8s.io/component-base/cli/flag"
+	"k8s.io/utils/ptr"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 )
 
+// The default values we use. Prometheus and Alertmanager are handled by
+// prometheus-operator. For thanos we use the default version from
+// prometheus-operator.
+var defaultImages = map[string]string{
+	"prometheus":   "",
+	"alertmanager": "",
+	"thanos":       obopo.DefaultThanosImage,
+}
+
+func imagesUsed() []string {
+	i := 0
+	imgs := make([]string, len(defaultImages))
+	for k := range defaultImages {
+		imgs[i] = k
+		i++
+	}
+	return imgs
+}
+
+// validateImages merges the passed images with the defaults and checks if any
+// unknown image names are passed. If an unknown image is found, this raises an
+// error.
+func validateImages(images *k8sflag.MapStringString) (map[string]string, error) {
+	res := defaultImages
+	if !images.Empty() {
```

Comment:
(nit) fast path if no custom images are provided
```suggestion
	if images.Empty() {
		return res
	}
```